developer/frameworks/articles/05-assets/05-relating-assets.markdown

```diff
@@ -0,0 +1,135 @@
+entity. This way only content creators can relate other assets to the entity.
+Related assets are implemented in the JSP by using the Liferay UI tag
+`liferay-ui:input-asset-links` inside a collapsible panel. This code is
+placed inside the `aui:fieldset` tags of the JSP. 
+
+1.  Add the `liferay-asset:input-asset-links` tag to your form. Here's how it's
+    added in the Blogs application: 
+
+        <aui:fieldset collapsed="<%= true %>" collapsible="<%= true %>" label="related-assets">
+				<liferay-asset:input-asset-links
+					className="<%= [AssetEntry].class.getName() %>"
+					classPK="<%= entryId %>"
+				/>
+		</aui:fieldset>
+
+    The following screenshot shows the Related Assets menu for an application. Note
+    that it is contained in a collapsible panel titled Related Assets.
+
+    ![Figure 1: Your portlet's entity is now available in the Related Assets *Select* menu.](../../images/related-assets-select-menu.png)
+
+2.  Unfortunately, the Related Assets menu shows your entity's fully qualified
+    class name. To replace it with a simplified name for your entity, add
+    a language key with the fully qualified class name for the key
+    and the name you want for the value. Put the language key in file
+    `docroot/WEB-INF/src/content/Language.properties` in your portlet. You can
+    refer to the 
+    [Overriding Language Keys](/docs/7-2/-/knowledge_base/overriding-language-keys)
+    tutorial for more documentation on using language properties.
+
+    Upon redeploying your portlet, the value you assigned to the fully qualified
+    class name in your `Language.properties` file shows in the Related Assets menu. 
+
+Awesome! Now content creators and editors can relate the assets of your
+application. The next thing you need to do is reveal any such related assets to
+the rest of your application's users. After all, you don't want to give everyone
+edit access just so they can view related assets!
+
+## Showing Related Assets
+
+You can show related assets in your application's view of that entity or, if
+you've implemented asset rendering for your custom entity, you can show related
+assets in the full content view of your entity for users to view in an Asset
+Publisher portlet.
+
+1.  You must get the `AssetEntry` object associated with your entity: 
+
+        <%
+        long insultId = ParamUtil.getLong(renderRequest, "insultId");
+        Insult ins = InsultLocalServiceUtil.getInsult(insultId);
+        AssetEntry assetEntry = AssetEntryLocalServiceUtil.getEntry(Insult.class.getName(), ins.getInsultId());
+        %>
+
+2.  Use the `liferay-asset:asset-links` tag to show the entity's related assets.
+    For this tag, you should retrieve the entity's class name and the variable
+    holding your instance object, so you can return its ID. The example code
+    below uses the example entity class `Insult` and an instance object
+    variable called `ins`:
```

Comment:
@stephenkostas, I don't see the entity class used in the example.

Reply:
I think that description slipped through the cracks on a couple updates. I just updated the branch with this text: 
" For this tag, you retrieve the `assetEntryId` from the `assetEntry` object, 
    retrieve your asset's `className`, and get the entity's primary key 
    (`classPK`) from the specific `entry`. The tag then retrieves any other 
    assets linked to your asset."

Reply:
Looks good @stephenkostas, thanks!

Reply:
Thanks!